### PR TITLE
Extract Organization

### PR DESCRIPTION
**Pull request by dbernheisel**
# What
Extract `Organization` schema singleton that contains all organizational information, such as email addresses, descriptions, logos, URLs, and so forth.

# Why
For any other communities to use this application, they'd need to fork the codebase, change the text, and then deploy. They would then be out-of-sync with any updates this app has to offer, but I've added instructions in the readme about how to catch up and get updates.

### CODE REVIEW

File: lib/factory.ex [hunk 1]
```diff
@@ -1,6 +1,20 @@
 defmodule Community.Factory do
   use ExMachina.Ecto, repo: Community.Repo
 
+  def organization_factory do
+    %Community.Organization{
```
Comment (drapergeek) [inline lib/factory.ex]: Is there a benefit to this being in the database? Would it make sense to have this be a config option or a module that you fill in?

File: test/plugs/assign_organization_test.exs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+defmodule Community.AssignOrganizationTest do
+  use Community.ConnCase, async: true
+  alias Community.AssignOrganization
+  alias Community.Organization
+
+  describe "call" do
+    test "assigns the organization" do
+      conn = AssignOrganization.call(%Plug.Conn{}, [])
+
+      assert %Organization{} = conn.assigns[:organization]
```
Comment (drapergeek) [inline test/plugs/assign_organization_test.exs]: It's unclear from this test where the `%Organization{}` that is assigned here comes from. It's expected that a test has nothing in the db to start.

File: web/queries/organization.ex [hunk 3]
```diff
@@ -0,0 +1,18 @@
+defmodule Community.Queries.Organization do
+  alias Community.Organization
+  alias Community.Repo
+
+  def find_organization do
+    Organization
+    |> Repo.one
+    |> case do
+      nil ->
+        %Organization{}
```
Comment (drapergeek) [inline web/queries/organization.ex]: This is an interesting decision. Can you explain your thoughts? This seems like the kind of thing that should be an exception was my thought but I'd love to hear why you went this route.

File: web/static/js/dribbble.js [hunk 4]
```diff
@@ -17,24 +17,25 @@ function composeDribbbleShot(originalElement, url, imageUrl) {
...
-  $.ajax({
-    url: dribbleBucketUrl({bucket: bucketId, token: accessToken, count: count}),
-    context: document.body
-  }).done(function(data) {
-    $.each(data, function(index, shot) {
-      console.log(shot);
-      let shotElement = composeDribbbleShot(basicElement, shot.html_url, shot.images.hidpi);
-      shotElement.show();
-      list.append(shotElement);
+  if(accessToken) {
```
Comment (drapergeek) [inline web/static/js/dribbble.js]: These changes look pretty unrelated to the organization. Should this be a different PR?

File: web/templates/home_page/recent_meetups.html.eex [hunk 5]
```diff
@@ -1,23 +1,23 @@
...
-      <a href="http://twitter.com/raleighdesignio">@raleighdesignio</a>
-      if you have one you'd like to add.
-    </p>
-  </div>
+<%= if @organization.upcoming_meetups_url do %>
+  <section class="section" data-meetup-url='<%= @organization.upcoming_meetups_url %>'>
+    <div class="section-description">
+      <h4>Upcoming Meetups</h4>
+      <p>
+        Meetups in and around <%= @organization.city %>. Shoot us a suggestion <%= twitter_link(@organization) %> if you have one you'd like to add.
```
Comment (drapergeek) [inline web/templates/home_page/recent_meetups.html.eex]: Does this run over 100 chars?

File: lib/factory.ex (hunk 1 above)
Comment (dbernheisel) [reply lib/factory.ex]: The goal is to have it persist outside of the app config, so other communities can deploy (and not fork) the codebase and update the particulars about their community.  If it's in a config option, then they'd have to fork.

File: web/static/js/dribbble.js (hunk 4 above)
Comment (dbernheisel) [reply web/static/js/dribbble.js]: Yep, I'll put it in another PR. Thanks.

File: web/queries/organization.ex (hunk 3 above)
Comment (dbernheisel) [reply web/queries/organization.ex]: When someone deploys their version of Community, they will have an empty database, and the app will crash. This will set them up with a placeholder Organization since the app now assumes there will be an organization.

Comment (dbernheisel) [reply web/queries/organization.ex]: I could instead put this into a mix task to prime the database for new deployments-- but this was also just as effective and didn't need another step.

File: lib/factory.ex (hunk 1 above)
Comment (drapergeek) [reply lib/factory.ex]: Could you accomplish the same by recommending a file name `organization_config.exs` and using:

```elixir
import_config "organization_config.exs"
```

File: web/templates/home_page/recent_meetups.html.eex (hunk 5 above)
Comment (dbernheisel) [reply web/templates/home_page/recent_meetups.html.eex]: ya. I cut it down to two lines now.

File: lib/factory.ex (hunk 1 above)
Comment (dbernheisel) [reply lib/factory.ex]: they'd still have to fork.

Comment (drapergeek) [reply lib/factory.ex]: No:

```elixir

config :community, :organization, %Organization{name: "Triangle Design"}

if File.exists?("config/organization_config.exs") do
  import_config "organization_config.exs"
end
```

This allows us to have a default they fall back on, no Git changes required and no db calls.

**Comment by dbernheisel** [comment]
@drapergeek I've taken another stab at it taking your `config.exs` suggestion. Could you take another look?

File: test/features/member_test.exs [hunk 6]
```diff
@@ -31,7 +31,7 @@ defmodule Community.Feature.MemberTest do
     assert_delivered_email Community.Email.admin_member_added(last_member)
   end
 
-  test "contacts a member that is available for hire" do
+  test "contacts a member that is available for hire", %{organization: _organization} do
```
Comment (drapergeek) [inline test/features/member_test.exs]: Do you have to use these? I think you can ignore these options when you don't need them.

Comment (dbernheisel) [reply test/features/member_test.exs]: Nope. I'll remove them. This was more of me remembering that it's available to use in feature and controller tests.
